codegen/field.go

```diff
@@ -95,6 +95,18 @@ func (b *builder) bindField(obj *Object, f *Field) error {
 		f.GoReceiverName = "ec"
 		f.GoFieldName = "introspectType"
 		return nil
+	case f.Name == "_entities":
```

Comment:
It would be nice to have a mechanism that can be extended with plugins.

Reply:
@vektah maybe if the user has an `_entities` query *without* federation...we can possibly check `cfg.Federated && f.Name == _entities` ... that said, it would be great if the plugin can inject this part, but it looks like there's no way right now...can you confirm? Thanks!

Reply:
Welcome back!

I wonder if we could add something to `TypeMapField` in config that replaces all of this logic?
```go
type TypeMapField struct {
	Resolver  bool   `yaml:"resolver"`
	FieldName string `yaml:"fieldName"`

	// new
	GeneratedMethod string `yaml:"-"`
}
```

which would be set [here](https://github.com/99designs/gqlgen/pull/851/files#diff-6edbafe93c80509bb043690f2a456213R104) to reference `__resolve_entities`

Reply:
@vektah that's an interesting idea...I'll give it a shot 👍

Reply:
Actually, TypeMapField is in `MutateConfig` not `MutateSchema` where you linked  above, can you expand more on how that would work? Thanks!

Reply:
Lets leave it like this for now and get this on master. Might do some cleanup once it lands to make this more pluggable.